database/opensearch/opensearch.go

```diff
@@ -0,0 +1,395 @@
+package opensearch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/opensearch-project/opensearch-go"
+)
+
+func init() {
+	database.Register("opensearch", &OpenSearch{})
+}
+
+const MigrationVersionDocId = "migration_version" // The id of a document used as a lock
+const DefaultTimeout = 1 * time.Minute            // Default operations timeout
+const DefaultIndex = ".migrations"                // Default index to handle migrations
```

Comment:
Would it be a good idea to give the flexibility to users to set the default index name?

Reply:
Hi, @digi0pz! It's possible by the standard conn string. `opensearch://user:password@host:port/index`